tests/components/igloohome/test_config_flow.py

```diff
@@ -0,0 +1,131 @@
+
+FORM_USER_INPUT = {
+    CONF_CLIENT_ID: "test-client-id",
+    CONF_CLIENT_SECRET: "test-client-secret",
+}
+
+
+async def test_form(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_auth: Generator[AsyncMock]
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        FORM_USER_INPUT,
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Client Credentials"
+    assert result["data"] == FORM_USER_INPUT
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.parametrize(
+    ("exception", "result_error"),
+    [(AuthException(), "invalid_auth"), (ClientError(), "cannot_connect")],
+)
+async def test_form_invalid_input(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_auth: Generator[AsyncMock],
+    exception: Exception,
+    result_error: str,
+) -> None:
+    """Tests where we handle errors in the config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_auth.side_effect = exception
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        FORM_USER_INPUT,
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": result_error}
+
+    # Make sure the config flow tests finish with either an
+    # FlowResultType.CREATE_ENTRY or FlowResultType.ABORT so
+    # we can show the config flow is able to recover from an error.
+    mock_auth.side_effect = None
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        FORM_USER_INPUT,
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Client Credentials"
+    assert result["data"] == FORM_USER_INPUT
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_abort_on_matching_entry(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_auth: Generator[AsyncMock]
+) -> None:
+    """Tests where we handle errors in the config flow."""
+    # Create first config flow.
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        FORM_USER_INPUT,
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Client Credentials"
+    assert result["data"] == FORM_USER_INPUT
+    assert len(mock_setup_entry.mock_calls) == 1
```

Comment:
```suggestion
setup_integration(hass, mock_config_entry)
```
you'll need `mock_config_entry` in the params and might need to remove `mock_setup_entry` from them (but please try with them first)

Reply:
Hey @zweckj , wasn't able to get it to work. The test failed because the entry was created despite the matching the user input and the fixture.